React/Modules/RCTUIManager.m

```diff
@@ -78,8 +78,15 @@ static UIViewAnimationOptions UIViewAnimationOptionsFromRCTAnimationType(RCTAnim
     case RCTAnimationTypeEaseInEaseOut:
       return UIViewAnimationOptionCurveEaseInOut;
     case RCTAnimationTypeKeyboard:
-      // http://stackoverflow.com/questions/18870447/how-to-use-the-default-ios7-uianimation-curve
+      // http://stackoverflow.com/questions/18870447/how-to-use-the-default-ios7-uianimation-curve/19439283#19439283
+      RCTLogWarn(@"The animation type RCTAnimationTypeKeyboard is deprecated. Instead use RCTAnimationTypeKeyboardShow or RCTAnimationTypeKeyboardHide.");
```

Comment:
Good idea, but can you move this deprecation warning to the JS side instead, that way the legacy support code can be kept only on JS side, making it easier to remove later.